src/core/radio/ble_secure.cpp

```diff
@@ -0,0 +1,542 @@
+            }
+
+            if (mReceivedMessage->GetLength() < requiredBytes)
+            {
+                continue;
+            }
+
+            // TLV fully loaded
+
+            if (mTcatAgent.IsEnabled())
+            {
+                ot::Message *message;
+                Error        error = kErrorNone;
+
+                message = Get<MessagePool>().Allocate(Message::kTypeBle);
+                VerifyOrExit(message != nullptr, error = kErrorNoBufs);
+
+                error = mTcatAgent.HandleSingleTlv(*mReceivedMessage, *message);
+                if (message->GetLength() != 0)
+                {
+                    IgnoreReturnValue(SendMessage(*message));
+                }
+
+                if (error == kErrorAbort)
+                {
+                    Disconnect(); // Keep BLE active (for testing)
+                    Stop();       // Stop BLE interface
+                    ExitNow();
+                }
+            }
+            else
+            {
+                mReceivedMessage->SetOffset((uint16_t)offset);
+                mReceiveCallback.InvokeIfSet(&GetInstance(), mReceivedMessage, (int32_t)offset,
+                                             OT_TCAT_MESSAGE_TYPE_NONE, "");
+            }
+
+            SuccessOrExit(mReceivedMessage->SetLength(0)); // also sets the offset to 0
+            requiredBytes = sizeof(Tlv);
+        }
+    }
+
+exit:
+    return;
+}
+
+void BleSecure::HandleTransmit(void)
+{
+    Error        error   = kErrorNone;
+    ot::Message *message = mTransmitQueue.GetHead();
+
+    VerifyOrExit(message != nullptr);
+    mTransmitQueue.Dequeue(*message);
+
+    if (mTransmitQueue.GetHead() != nullptr)
+    {
+        mTransmitTask.Post();
+    }
+
+    SuccessOrExit(error = mTls.Send(*message, message->GetLength()));
+
+exit:
+    if (error != kErrorNone)
+    {
+        LogNote("Transmit: %s", ErrorToString(error));
+        message->Free();
+    }
+    else
+    {
+        LogDebg("Transmit: %s", ErrorToString(error));
+    }
+}
+
+Error BleSecure::HandleTransport(void *aContext, ot::Message &aMessage, const Ip6::MessageInfo &aMessageInfo)
+{
+    OT_UNUSED_VARIABLE(aMessageInfo);
+    return static_cast<BleSecure *>(aContext)->HandleTransport(aMessage);
+}
+
+Error BleSecure::HandleTransport(ot::Message &aMessage)
+{
+    otBleRadioPacket packet;
+    uint16_t         len    = aMessage.GetLength();
+    uint16_t         offset = 0;
+    Error            error  = kErrorNone;
+
+    while (len > 0)
+    {
+        if (len <= mMtuSize - kGattOverhead)
+        {
+            packet.mLength = len;
+        }
+        else
+        {
+            packet.mLength = mMtuSize - kGattOverhead;
+        }
+
+        if (packet.mLength > kPacketBufferSize)
+        {
+            packet.mLength = kPacketBufferSize;
+        }
+
+        IgnoreReturnValue(aMessage.Read(offset, mPacketBuffer, packet.mLength));
+        packet.mValue = mPacketBuffer;
+        packet.mPower = 0;
```

Comment:
Should Tx power be configurable, or is there a requirement that always forces this to 0 dBm ? A build-time constant could be used, or a runtime configurable parameter. (May also be done in later PRs I think - if we believe 0 dBm is good enough to get started. In this case  a TODO: note should be added here.)

Reply:
@canisLupus1313 introduce constant and TODO clause.